source/MaterialXGenShader/HwShaderGenerator.cpp

```diff
@@ -93,20 +94,31 @@ ShaderPtr HwShaderGenerator::createShader(const string& name, ElementPtr element
     // Add a block for data from vertex to pixel shader.
     addStageConnectorBlock(HW::VERTEX_DATA, "vd", *vs, *ps);
 
-    // Create uniforms for environment lighting
-    // Note: Generation of the rotation matrix using floating point math can result
-    // in values which when output can't be consumed by a h/w shader, so
-    // just setting explicit values here for now since the matrix is simple.
-    // In general the values will need to be "sanitized" for hardware.
-    const Matrix44 yRotationPI(-1, 0, 0, 0,
-        0, 1, 0, 0,
-        0, 0, -1, 0,
-        0, 0, 0, 1);
-    psPrivateUniforms->add(Type::MATRIX44, "u_envMatrix", Value::createValue<Matrix44>(yRotationPI));
-    psPrivateUniforms->add(Type::FILENAME, "u_envIrradiance");
-    psPrivateUniforms->add(Type::FILENAME, "u_envRadiance");
-    psPrivateUniforms->add(Type::INTEGER, "u_envRadianceMips", Value::createValue<int>(1));
-    psPrivateUniforms->add(Type::INTEGER, "u_envSamples", Value::createValue<int>(16));
+    // Disable specular environment code if lighting is not required.
+    bool lighting = graph->hasClassification(ShaderNode::Classification::SHADER | ShaderNode::Classification::SURFACE) ||
+                    graph->hasClassification(ShaderNode::Classification::BSDF);
+    if (!lighting)
+    {
+        context.getOptions().hwSpecularEnvironmentMethod = SPECULAR_ENVIRONMENT_NONE;
```

Comment:
We should avoid changing the context options like this. We don't know if the context is saved and reused on client side, and this could affect successive runs of shader generation. The fact that you _can_ change option values like this feels like a design flaw, and we should log an issue for that.

Reply:
I didn't think about the side-effect as I thought this was a copy. I can just test the lighting flag and not change the options.